derive/src/pystructseq.rs

```diff
@@ -38,7 +46,24 @@ pub(crate) fn impl_pystruct_sequence(input: DeriveInput) -> Result<TokenStream>
         }
         impl ::rustpython_vm::convert::ToPyObject for #ty {
             fn to_pyobject(self, vm: &::rustpython_vm::VirtualMachine) -> ::rustpython_vm::PyObjectRef {
-                ::rustpython_vm::PyStructSequence::into_struct_sequence(self, vm).into()
+                ::rustpython_vm::types::PyStructSequence::into_struct_sequence(self, vm).into()
+            }
+        }
+    };
+    Ok(ret)
+}
+
+pub(crate) fn impl_pystruct_sequence_try_from_object(input: DeriveInput) -> Result<TokenStream> {
+    let field_names = field_names(&input)?;
+    let ty = &input.ident;
+    let ret = quote! {
+        impl ::rustpython_vm::TryFromObject for #ty {
+            fn try_from_object(vm: &::rustpython_vm::VirtualMachine, seq: ::rustpython_vm::PyObjectRef) -> ::rustpython_vm::PyResult<Self> {
+                let seq = Self::try_elements_from(seq, vm)?;
+                let mut iter = seq.into_iter();
```

Comment:
Yeah, this could probably be `let seq: [PyObjectRef; <#ty as ::rustpython_vm::types::PyStructSeq>::FIELD_NAMES.len()] = seq.try_into().unwrap();`

Reply:
no idea how to handle index in `quote!()`